### PR TITLE
Boil 43 template params

### PR DESCRIPTION
**Pull request by ashtonmeuser**
Align OpenShift objects names
Use parameter for OpenShift build branch
Fix DB config template indentation
Use database secret in server
Option to randomly generate DB user and password
Remove annotations used for template brokers

### CODE REVIEW

**Comment by ashtonmeuser** [comment]
Use this to deploy replacing the URL with one relevant to your Minishift instance.

```
make deploy-database NAMESPACE=launchpad POSTGRESQL_DATABASE=sample_db 

make deploy-server NAMESPACE=launchpad BRANCH=BOIL-43-template-params

make deploy-client NAMESPACE=launchpad BRANCH=BOIL-43-template-params API_URL=http://openshift-launchpad-server-launchpad.192.168.64.4.nip.io
```

File: deployment/client.bc.json [hunk 1]
```diff
@@ -2,49 +2,56 @@
...
+            "name": "${APP_NAME}-client:latest",
             "namespace": "${NAMESPACE}"
           }
         },
         "source": {
           "contextDir": "client",
           "git": {
-            "ref": "origin/deployment-v2",
+            "ref": "origin/${BRANCH}",
             "uri": "https://github.com/bcgov/openshift-launchpad"
```
Comment (ghost) [inline deployment/client.bc.json]: Should this be an env variable too?

File: Makefile [hunk 2]
```diff
@@ -32,66 +32,54 @@ client-test:
...
-# Server commands
-############################
+##############################################################################
+# Deployment cleanup commands
+##############################################################################
+
+oc-all-clean:
+	@echo "+\n++ Tearing down all OpenShift objects created from templates...\n+"
+	@oc delete all -l app=openshift-launchpad
+	@oc delete pvc openshift-launchpad-database
```
Comment (hitankar-freshworks) [inline Makefile]: Maybe not a good idea for including 'persistent volume claim'. Data in the storage will get removed if the team decides to flush their configs.

Comment (hitankar-freshworks) [reply Makefile]: @andrewwhitehead how is this delete pvc different from oc-persisted-clean?

**Comment by ashtonmeuser** [comment]
New deployment commands (change `API_URL`).

```
make deploy-database NAMESPACE=launchpad APP_NAME=openshift-launchpad POSTGRESQL_DATABASE=launchpad

make deploy-server NAMESPACE=launchpad APP_NAME=openshift-launchpad REPO=https://github.com/bcgov/openshift-launchpad BRANCH=BOIL-43-template-params

make deploy-client NAMESPACE=launchpad APP_NAME=openshift-launchpad REPO=https://github.com/bcgov/openshift-launchpad BRANCH=BOIL-43-template-params API_URL=http://openshift-launchpad-server-launchpad.192.168.64.4.nip.io
```

**Review by hitankar-freshworks — APPROVED** [review]

**Pull request merged**